### PR TITLE
Add bpf metics plugin

### PR DESCRIPTION
**Pull request by HeinrichHartmann**
This PR adds support for collecting:
- block IO latencies
- runqueue latencies and
- syscall latencies
as histogram metrics on Linux systems with kernel versions 4.3+ (in particular Ubuntu 16.04).

This branch includes changes from PR #108.

### CODE REVIEW

File: plugins/linux/bccbpf/Makefile [hunk 1]
```diff
@@ -1,7 +1,7 @@
...
 
-all: iolatency.elf
+all: bpf.elf iolatency.elf
 
-iolatency.elf: iolatency.c
-	$(CC) iolatency.c -o iolatency.elf
-	chmod +s iolatency.elf
+%.elf : %.c
+	$(CC) $< -o $@
+	chmod +s $@
```
Comment (HeinrichHartmann) [inline plugins/linux/bccbpf/Makefile]: Creating a setuid executable is security risk. Shall we automate this step, or leave it to the user to do this manually.

File: plugins/linux/bccbpf/lua/circll.lua [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+static circll_bin_t circll_bin(u64 v, s8 exp_offset) {
+  s8 exp = 1;
+  if(v == 0) return (circll_bin_t)  {.val = 0,    .exp = 0 + exp_offset};
+  if(v < 10) return (circll_bin_t)  {.val = v*10, .exp = 1 + exp_offset};
+  LLN128()
+  if(v > 100) return (circll_bin_t) {.val = -1,   .exp = 0 + exp_offset};
+ good:
+  return (circll_bin_t) {.val = v, .exp = exp + exp_offset};
+}
+]]
```
Comment (HeinrichHartmann) [inline plugins/linux/bccbpf/lua/circll.lua]: I was not able to get teh exp_offset working, with the bit-shifting tricks used in iolatency.py. So I avoided those by using the bucket struct from libcircllhist. Do you see any drawbacks, @postwait ?

Comment (postwait) [inline plugins/linux/bccbpf/lua/circll.lua]: I do not see drawbacks. Same spirit.

File: plugins/linux/bccbpf/iolatency.py [hunk 3]
```diff
@@ -78,32 +77,14 @@
...
-        description="Summarize block device I/O latency as a histogram",
-        formatter_class=argparse.RawDescriptionHelpFormatter,
-        epilog=examples)
-    parser.add_argument(
-        "-Q",
-        "--queued",
-        action="store_true",
-        help="include OS queued time in I/O time")
-    args = parser.parse_args()
+    # don't blog the nad process at startup
```
Comment (postwait) [inline plugins/linux/bccbpf/iolatency.py]: Block

**Comment by HeinrichHartmann** [comment]
- Removed `chmod +s` from Makefile, to avoid installing setuid executables on all hosts running nad.
- Provide copy-paste-able installation instructions for Ubuntu 16

**Comment by HeinrichHartmann** [comment]
- [x] Tested omnibus install on fresh Vagrant boxes: u14, u16, c6, c7.
- [x] Tested bpf install instructions on u16.

**Comment by HeinrichHartmann** [comment]
Got the OK from @maier to merge this.

**Pull request merged**